packages/sql-pg/src/PgClient.ts

```diff
@@ -64,6 +63,27 @@ export interface PgClientConfig {
   readonly username?: string | undefined
   readonly password?: Redacted.Redacted | undefined
 
+  /**
+   * A function returning a custom socket to use. This parameter is not documented
+   * in the postgres.js's type signature. See their
+   * [readme](https://github.com/porsager/postgres?tab=readme-ov-file#connection-details) instead.
+   *
+   * @example
+   * import { AuthTypes, Connector } from "@google-cloud/cloud-sql-connector";
+   * import { PgClient } from "@effect/sql-pg";
+   * import { Effect, Layer } from "effect"
+   *
+   * const layer = Effect.gen(function*() {
+   *   const connector = new Connector();
+   *   const clientOpts = yield* Effect.promise(() => connector.getOptions({
+   *     instanceConnectionName: "project:region:instance",
+   *     authType: AuthTypes.IAM,
+   *   }));
+   *   return PgClient.layer({ socket: clientOpts.stream, user: "iam-user" });
+   * }).pipe(Layer.unwrapEffect)
+   */
+  readonly socket?: (() => unknown) | undefined
```

Comment:
Can unknown be replaced with a node.js Duplex?

Reply:
Honestly, I wasn't sure which one of `TLSSocket`, `Socket`, `Duplex`,... to constrain it with and hoped for your input. I'll fix it. Thanks.